crates/router/src/connector/braintree/transformers.rs

```diff
@@ -254,234 +549,1255 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
             BraintreePaymentStatus::Authorized => Self::Authorized,
             BraintreePaymentStatus::Voided => Self::Voided,
             BraintreePaymentStatus::SubmittedForSettlement
-            | BraintreePaymentStatus::SettlementPending
-            | BraintreePaymentStatus::SettlementConfirmed => Self::Pending,
+            | BraintreePaymentStatus::SettlementPending => Self::Pending,
         }
     }
 }
 
-impl<F, T>
-    TryFrom<types::ResponseRouterData<F, BraintreePaymentsResponse, T, types::PaymentsResponseData>>
-    for types::RouterData<F, T, types::PaymentsResponseData>
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            BraintreePaymentsResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
         item: types::ResponseRouterData<
             F,
             BraintreePaymentsResponse,
-            T,
+            types::PaymentsAuthorizeData,
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
-        let id = item.response.transaction.id.clone();
-        Ok(Self {
-            status: enums::AttemptStatus::from(item.response.transaction.status),
-            response: Ok(types::PaymentsResponseData::TransactionResponse {
-                resource_id: types::ResponseId::ConnectorTransactionId(id.clone()),
-                redirection_data: None,
-                mandate_reference: None,
-                connector_metadata: None,
-                network_txn_id: None,
-                connector_response_reference_id: Some(id),
-                incremental_authorization_allowed: None,
-                charge_id: None,
+        match item.response {
+            BraintreePaymentsResponse::ErrorResponse(error_response) => Ok(Self {
+                response: build_error_response(&error_response.errors.clone(), item.http_code),
+                ..item.data
             }),
-            ..item.data
-        })
+            BraintreePaymentsResponse::PaymentsResponse(payment_response) => {
+                let transaction_data = payment_response.data.charge_credit_card.transaction;
+                let status = enums::AttemptStatus::from(transaction_data.status.clone());
+                let response = if let enums::AttemptStatus::Failure = status {
+                    Err(types::ErrorResponse {
+                        code: transaction_data.status.to_string().clone(),
+                        message: transaction_data.status.to_string().clone(),
+                        reason: None,
+                        attempt_status: None,
+                        connector_transaction_id: None,
```

Comment:
Can we map `transaction_data.id` to  `connector_transaction_id` here ?

Reply:
mapped